recipes-core/platform-onl-init/files/platform-onl-init.sh

```diff
@@ -5,18 +5,26 @@ add_port() {
 	local portnum=$2
 	local i2c_bus_no=$3
 
-	create_i2c_dev ${i2c_dev_name} 0x50 ${i2c_bus_no}
-	wait_for_file "/sys/bus/i2c/devices/${i2c_bus_no}-0050/port_name"
-	echo "port${portnum}" > "/sys/bus/i2c/devices/${i2c_bus_no}-0050/port_name"
+	create_i2c_dev ${i2c_dev_name} 0x50 ${i2c_bus_no} "port_name=port${portnum}"
 }
 
 create_i2c_dev() {
 	local i2c_dev_name=$1
 	local i2c_addr=$2
 	local i2c_bus_no=$3
+	local i2c_addr_4="00${i2c_addr:2}"
 
 	wait_for_file "/sys/bus/i2c/devices/i2c-${i2c_bus_no}/new_device"
 	echo ${i2c_dev_name} ${i2c_addr} > /sys/bus/i2c/devices/i2c-${i2c_bus_no}/new_device
+
+	# apply additional parameters passed as <parameter>=<value> strings
+	while [ -n "$4" ]; do
```

Comment:
```suggestion
	while [ -n "${4:-""}" ]; do
```

Fix "/usr/bin/platform-onl-init.sh: line 21: $4: unbound variable".

Reply:
where does this come from?

Reply:
accton-as5835-54x-1. Reboot and see it fail.

Reply:
I see.

Anyway, I pushed a (slightly) different fix.

Reply:
Elegant. I like it.